### ISSUE TITLE
Can't be downloaded?!

### ISSUE THREAD

**Issue by Skylineman** [KraigM/homebridge-nest#139]
Hello,

I'm getting the following issues when I'm trying to install any of the nest packages.
`npm ERR! code 128
npm ERR! Command failed: /usr/bin/git clone -q https://github.com/kraigm/unofficial_nodejs_nest.git /root/.npm/_cacache/tmp/git-clone-993140f0
npm ERR! fatal: could not create leading directories of '/root/.npm/_cacache/tmp/git-clone-993140f0': Permission denied`

I have looked around and tried all the given solutions over the net, but it's something only with the nest packages. Any other is installed easily.

Do you happen to have an alternate solution to install?

Thanks,
Marcell

**Comment by gerarddvb** [comment]
Add ‘sudo’ to npm install -g

**Comment by bassrock** [comment]
Im getting the same issue. I also tried to install with yarn, that did succeed, however it is not getting put in the node_modules folder.

**Comment by roe1974** [comment]
any solution ?

**Comment by DJay-X** [comment]
Same problem here.

sudo npm install -g homebridge-nest

```
npm ERR! code 128
npm ERR! Command failed: /usr/bin/git clone -q https://github.com/kraigm/unofficial_nodejs_nest.git /root/.npm/_cacache/tmp/git-clone-0c8b9505
npm ERR! fatal: Konnte führende Verzeichnisse von '/root/.npm/_cacache/tmp/git-clone-0c8b9505' nicht erstellen.: Keine Berechtigung
npm ERR! 

npm ERR! A complete log of this run can be found in:
npm ERR!     /root/.npm/_logs/2017-10-01T06_39_43_505Z-debug.log
```

**Comment by benfysh** [comment]
I am also seeing the same issue. Other plugins are working fine.

`npm ERR! Command failed: /usr/bin/git clone -q https://github.com/kraigm/unofficial_nodejs_nest.git /root/.npm/_cacache/tmp/git-clone-f9f314f6
npm ERR! fatal: could not create leading directories of '/root/.npm/_cacache/tmp/git-clone-f9f314f6': Permission denied`

**Comment by jocamero** [comment]
Same issue when attempting to update from 1.1.2 to 1.1.3.

Tried:
sudo npm install -g homebridge-nest
&
sudo npm update -g homebridge-nest

other plugins and homebridge updated without issue.

Anyone know what might be causing this or how to fix it?

**Comment by roe1974** [comment]
i think its due to an error in 
https://github.com/kraigm/unofficial_nodejs_nest.git
only @KraigM can solve this ....

**Comment by roe1974** [comment]
anyone tried:

sudo chown -R $USER /usr/local && npm install homebridge-nest -g

??

**Comment by KraigM** [comment]
You guys are getting directory permission issues.  It might be related to the fact that that dependency has to be pulled from GitHub due to the fact that the fix I submitted a long time ago has still not been released to npm.

**Comment by roe1974** [comment]
so is there a workaround for us ?

**Comment by KraigM** [comment]
Now technically that dependency is not needed anymore, if you are using the new API stuff.  It is only kept around for legacy purposes (version zero users).

**Comment by roe1974** [comment]
ok, but the dependency is in the package.json ....

**Comment by KraigM** [comment]
Yes... because if I remove it then anyone still using the old way would break due to a missing package

**Comment by KraigM** [comment]
Personally, I strongly suggest switching to [nvm](https://github.com/creationix/nvm) as many problems like this (file permission issues) are handled.  **IMPORTANT: After switching, you no longer use `sudo` in any of the commands** 

Switching does unfortunately require you to reinstall homebridge/plugins but it will use your same config as it did before

**Comment by roe1974** [comment]
hmmmm .... sounds not so good.. because homebridge and all plugins are working now perfect...
no other way ? ;-)
... and users still using the old way would break during update/install ... or while the plugin is running ?

... and NVM is not even simple to install ;-)

**Comment by KraigM** [comment]
Another solution/workaround may be to follow npm's guide on permission issues:
https://docs.npmjs.com/getting-started/fixing-npm-permissions

It sounds like there is a permission issue on the folder '/root/.npm/' (as it specifically says on the error).  What happens when you run `npm config get prefix`?

**Comment by roe1974** [comment]
/usr
so you mean try option 2 from the link........?

**Comment by KraigM** [comment]
Seems to be an issue with npm: https://github.com/npm/npm/issues/16898

**Comment by roe1974** [comment]
can you change the package.json for 1-2h ... just for installing it ? ... or is there a problem later ?

**Comment by KraigM** [comment]
Not really.  For one, it is not as simple as removing the dependency since the legacy functionality is still being loaded atm.  Also, npm doesn't really work like that.

**Comment by KraigM** [comment]
But I'm nice and I removed the legacy handling and tested it for you.  Since you can't pull it from git, I released it as version 1.1.3-nolegacy. You can install using:

```
sudo npm install -g homebridge-nest@1.1.3-nolegacy 
```

Since this is not an actual issue, but just a work around for your file permission issues, I do not want to officially release this without making it a major version (like 2.0.0).

**Comment by roe1974** [comment]
perfect !!!
install works ;-)

**Comment by travmo** [comment]
I am so thankful for your work KraigM. Been struggling with this as well and glad to see a workaround at least for the time being.

**Comment by roe1974** [comment]
my Nest sensors are back to homekit :-)))))))))
@KraigM ..... anyway to donate for your project/work ?

regards, richard
